api/v1alpha1/tls_types.go

```diff
@@ -19,6 +22,20 @@ type ClientTLSSettings struct {
 	// Session defines settings related to TLS session management.
 	// +optional
 	Session *Session `json:"session,omitempty"`
+
+	// OCSPStaple references a Kubernetes ConfigMap or Secret containing the OCSP stapled response
+	// that will be presented by Envoy during TLS handshakes.
+	//
+	// The referenced object must contain a key named `ocsp.staple` with the
+	// base64-encoded OCSP response.
+	//
+	// Envoy Gateway also supports `ocsp.staple` in the certificateRefs of a Gateway Listener.
+	//
+	// If both `OCSPStaple` in ClientTrafficPolicy and `ocsp.staple` in certificateRefs are configured,
+	// the value from ClientTrafficPolicy takes precedence.
+	//
+	// +optional
+	OCSPStaple *gwapiv1.SecretObjectReference `json:"ocspStaple,omitempty"`
```

Comment:
I dont think this is needed, the use case is for sTLS

Reply:
After discussed with @arkodg offline, remove the separate ocsp staple and use the server tls secrect for now.